api/peripheral_api/util/RFID-helpers.js

```diff
@@ -0,0 +1,195 @@
+   */
+  testing() {
+    return this.inTesting;
+  }
+
+  /**
+   * Helper function to hash the RFID data for security purposes
+   * @param {String|null} byte RFID card data
+   * @returns a hashed value if successful else null
+   */
+  hashedCardData(byte) {
+    return new Promise((resolve, reject) => {
+      bcrypt.genSalt(10, function(error, salt) {
+        if (error) {
+          resolve(null);
+        }
+        bcrypt.hash(byte, secretKey, function(hashError, hash) {
+          if (hashError) {
+            resolve(null);
+          }
+          resolve(hash);
+        });
+      });
+    });
+  }
+
+  /**
+   * Retrieves all RFID cards in the database
+   * @returns {Promise<Array<Object>>}
+   */
+  getCards() {
+    return new Promise((resolve, reject) => {
+      RFID.find()
+        .then((items) => resolve(items))
+        .catch((error) => {
+          resolve([]);
+        });
+    });
+  }
+
+  /**
+   * Validates the scanned RFID card by comparing the hashed data
+   * with the one in the database
+   * @param {JSONObject} payload Incoming card data from ESP32
+   * @returns {Number} 404 if card data was null or not in the database
+   * else 200 if the incoming data is not null and it exists in the database
+   */
+  async validate(payload) {
+    return new Promise(async (resolve, reject) => {
+      const { message } = JSON.parse(payload.toString());
+      const hash = await this.hashedCardData(message);
+      if (hash === null) {
+        resolve(NOT_FOUND);
+      } else {
+        let cards = [];
+        cards.push(... await this.getCards());
+        let found = false;
+        for (let card of cards) {
+          if (card.byte === hash) {
+            found = true;
+            resolve(OK);
+          }
+        }
+        if (!found) {
+          resolve(NOT_FOUND);
+        }
+      }
+
+    });
+  }
+
+  /**
+   * Creates and adds a new RFID card in the database
+   * @param {String|null} name Name of the cardholder
+   * @param {JSONObject} payload Incoming card data from ESP32
+   * @returns {Number} 400 if incoming data was null or if the card already
+   * exists in the database else returns 200 if successful
+   */
+  async createRfid(name, payload) {
+    return new Promise(async (resolve, reject) => {
+      const newRFID = new RFID({
+        name: name,
+        byte: await this.hashedCardData(JSON.parse(payload.toString()).message)
+      });
+      if (newRFID.name === null || newRFID.byte === null) {
+        resolve(BAD_REQUEST);
+      } else {
+        RFID.create(newRFID, (error) => {
+          if (error) {
+            resolve(BAD_REQUEST);
+          } else {
+            resolve(OK);
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * @param {Wrapper} device  MQTT Client Wrapper
+   * @param {JSONObject} payload Incoming card data from ESP32
+   * Handles incoming messages from ESP32
+   * If in creating mode, then it will create new RFID card
+   * else validate RFID card
```

Comment:
```suggestion
   * Handles incoming messages from ESP32
   * If in creating mode, then it will create new RFID card
   * else validate RFID card
   * @param {Wrapper} device  MQTT Client Wrapper
   * @param {JSONObject} payload Incoming card data from ESP32
```